_source/_posts/2022-10-20-spring-vault.md

```diff
@@ -0,0 +1,642 @@
+</code></pre>
+
+Using environment variables for passing secrets to containerized applications is now considered bad practice because the environment can be inspected or logged in a number of cases. So, let's move on to using Spring Cloud Config server for secrets storage.
+
+### Use Auth0 for OpenID Connect
+
+For using Auth0 as OIDC provider, you need to add the `spring-boot-starter-oauth2-client` dependency, as the Okta Spring Boot Starter [does not support Auth0](https://github.com/okta/okta-spring-boot/issues/358) yet.
+
+You can create a demo application with Spring Initializr too:
+
+```shell
+https start.spring.io/starter.zip \
+  bootVersion==2.7.4 \
+  dependencies==web,oauth2-client,cloud-config-client \
+  groupId==com.okta.developer \
+  artifactId==vault-demo-app-auth0  \
+  name=="Spring Boot Application" \
+  description=="Demo project of a Spring Boot application with Vault protected secrets" \
+  packageName==com.okta.developer.vault > vault-demo-app-auth0.zip
+```
+
+Modify the `Application` class the same way as described in the previous section. Also set `spring.cloud.config.enabled=false` in `application.properties`.
+
+Sign up at [Auth0](https://auth0.com/signup) and install the [Auth0 CLI](https://github.com/auth0/auth0-cli). Then run:
+
+```shell
+auth0 login
+```
+
+The terminal will display a device confirmation code and open a browser session to activate the device.
+
+**NOTE**: My browser was not displaying anything, so I had to manually activate the device by opening the URL https://auth0.auth0.com/activate?user_code={deviceCode}
+
+On successful login, you will see the tenant, which you will use as issuer later:
+
+```shell
+✪ Welcome to the Auth0 CLI 🎊
+
+If you don't have an account, please create one here: https://auth0.com/signup.
+
+Your device confirmation code is: KGFL-LNVB
+
+ ▸    Press Enter to open the browser to log in or ^C to quit...
+
+Waiting for the login to complete in the browser... ⣻Opening in existing browser session.
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-avup2laz.us.auth0.com
+```
```

Comment:
The coloring looks funny on this after the first '. I'm not sure how to fix it.

Reply:
Fixed, removed the `shell` highlighting.